provisioning/transport/http/src/http.ts

```diff
@@ -0,0 +1,251 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+'use strict';
+
+import { EventEmitter } from 'events';
+import { RestApiClient, Http as Base } from 'azure-iot-http-base';
+import { errors, X509 } from 'azure-iot-common';
+import * as machina from 'machina';
+
+export type ProvisioningResponseCallback = (err: Error, result?: any) => void;
+
+const _defaultHeaders = {
+  'Accept' : 'application/json',
+  'Content-Type' : 'application/json; charset=utf-8'
+};
+
+export class Http extends EventEmitter {
+
+  private _idScope: string;
+  private _registrationId: string;
+  private _body: any;
+  private _forceRegistration: boolean;
+  private _operationId: string;
+
+  private _fsm: machina.Fsm;
+  private _apiVersion: string = '2017-08-31-preview';
+  private _provisioningServer: string = 'global.azure-devices-provisioning.net';
```

Comment:
>'global.azure-devices-provisioning.net' [](start = 40, length = 39)

Do I remember some discussion about not having URLs hard coded just in case to support all clouds (public and not?) need to double check.

Reply:
Yes, will add a config object as part of the interface

---
In reply to: [142297735](https://github.com/Azure/azure-iot-sdk-node/pull/99#discussion_r142297735) [](ancestors = 142297735)